sycl/plugins/esimd_cpu/CMakeLists.txt

```diff
@@ -2,9 +2,120 @@
+    BUILD_COMMAND     cd ${CMAKE_CURRENT_BINARY_DIR}/libva-prefix/src/libva && make -j
+    INSTALL_COMMAND   cd ${CMAKE_CURRENT_BINARY_DIR}/libva-prefix/src/libva && make install
+  )
+  ExternalProject_Add_Step(libva llvminstall
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${LLVM_BINARY_DIR}/pi_esimd_cpu_deps && ${CMAKE_COMMAND} -E copy_directory <INSTALL_DIR>/ ${LLVM_BINARY_DIR}/pi_esimd_cpu_deps
+    COMMENT "Installing libva into the LLVM binary directory"
+    DEPENDEES install
+  )
+endif()
+
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_build)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install)
+
+if (MSVC)
+  set(LIBCM ${LLVM_BINARY_DIR}/pi_esimd_cpu_deps/lib/libcm${CMAKE_STATIC_LIBRARY_SUFFIX})
+  set(LIBIGFXCMRT_EMU ${LLVM_BINARY_DIR}/pi_esimd_cpu_deps/lib/igfxcmrt64_emu${CMAKE_STATIC_LIBRARY_SUFFIX})
+else()
+  set(LIBCM ${LLVM_BINARY_DIR}/pi_esimd_cpu_deps/lib/libcm${CMAKE_SHARED_LIBRARY_SUFFIX})
+  set(LIBIGFXCMRT_EMU ${LLVM_BINARY_DIR}/pi_esimd_cpu_deps/lib/libigfxcmrt_emu${CMAKE_SHARED_LIBRARY_SUFFIX})
+endif()
+
+if (DEFINED CM_LOCAL_SOURCE_DIR)
+    # Using local CM directory for online building without downloading
+  if (MSVC)
+    ExternalProject_Add(cm-emu
+      DOWNLOAD_COMMAND  ""
+      SOURCE_DIR        ${CM_LOCAL_SOURCE_DIR}
+      BINARY_DIR        ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_build
+      INSTALL_DIR       ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install
+      CMAKE_ARGS        -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+      BUILD_BYPRODUCTS  ${LIBCM} ${LIBIGFXCMRT_EMU}
+    )
+  else()
+    ExternalProject_Add(cm-emu
+      DOWNLOAD_COMMAND  ""
+      SOURCE_DIR        ${CM_LOCAL_SOURCE_DIR}
+      BINARY_DIR        ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_build
+      INSTALL_DIR       ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install
+      CMAKE_ARGS        -DLIBVA_INSTALL_PATH=${CMAKE_CURRENT_BINARY_DIR}/libva_install
+                        -D__SYCL_EXPLICIT_SIMD_PLUGIN__=true
+                        -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+      DEPENDS           libva
+      BUILD_BYPRODUCTS  ${LIBCM} ${LIBIGFXCMRT_EMU}
+    )
+  endif()
+else ()
+  if (DEFINED CM_PACKAGE_URL)
+    # Downloading pre-built CM Package
+    file (MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install)
+    ExternalProject_Add(cm-emu
+      URL               ${CM_PACKAGE_URL}
+      CONFIGURE_COMMAND ""
+      BUILD_COMMAND     ""
+      UPDATE_COMMAND    ""
+      INSTALL_DIR       ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install
+      INSTALL_COMMAND   ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_BINARY_DIR}/cm-emu-prefix/src/cm-emu/ <INSTALL_DIR>
+      BUILD_BYPRODUCTS  ${LIBCM} ${LIBIGFXCMRT_EMU}
+    )
+  else()
+    # Build from CM source tree fetched from github
+    if (MSVC)
+      message(FATAL_ERROR "Online-building of CM_EMU library is not supported under Windows environment")
```

Comment:
@kbobrovs , buildbot jobs for Windows environment will likely fail as CM_EMU is not supported yet for Windows. Can I apply `--disable-esimd-cpu` command line option to `configure.py` for Windows environment buildbot jobs? The option will prevent ESIMD_CPU module from building (ref : `sycl/plugins/CMakeLists.txt`, line 18)

Reply:
I changed CMake files to excluded pi_esimd_cpu / esimd_cpu building under Windows environment.